internal/pkg/externalplugins/tars/tars.go

```diff
@@ -241,13 +241,17 @@ func HandlePushEvent(log *logrus.Entry, ghc githubClient, pe *github.PushEvent,
 			"pr":   num,
 		})
 
-		// Only one PR is processed at a time, because even if other PRs are updated,
-		// they still need to be queued for another update and merge.
-		// To save testing resources we only process one PR at a time.
+		// Skips PRs with conflicting or unknown status.
+		if pr.Mergeable != githubql.MergeableStateMergeable {
+			continue
+		}
 		err = handle(l, ghc, &pr, cfg)
 		if err != nil {
 			l.WithError(err).Error("Error handling PR.")
 		} else {
+			// Only one PR is processed at a time, because even if other PRs are updated,
+			// they still need to be queued for another update and merge.
+			// To save testing resources we only process one PR at a time.
```

Comment:
I still don't quite understand this place. Can you explain when other PR skipped by break will be processed?

Reply:
The skipped pr will be processed in the half-hourly timed scan.

Reply:
So in the push event, tars will only process the earliest created, non-conflicting, open PR in the same branch.

Reply:
Yes.

Reply:
What are the benefits of this?

Reply:
I think I've described it in the documentation because we can pick a pr to update when there are new commits, rather than waiting for all the scans. This change will be helpful to speed up merging. Because it goes from a passive scan to a push.

Reply:
Ok, got it. 

Another question is, will there be a situation where the handle of a push event always handles the same one PR?

Reply:
Good catch! Still have some issue. I will fix it.

Reply:
Fixed.